src/NuGet.Jobs.GitHubIndexer/DiskRepositoriesCache.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using NuGetGallery;
+
+namespace NuGet.Jobs.GitHubIndexer
+{
+    public class DiskRepositoriesCache : IRepositoriesCache
+    {
+        private const string CacheFileSuffix = "-Cache.json";
+        private readonly ILogger<DiskRepositoriesCache> _logger;
+        public DiskRepositoriesCache(ILogger<DiskRepositoriesCache> logger)
+        {
+            _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+        }
+
+        /// <summary>
+        /// Creates a cache file for the specified repo that contains the repo's dependencies.
+        /// </summary>
+        /// <param name="repo">Repo to persist on disk</param>
+        public void Persist(RepositoryInformation repo)
+        {
+            var repoFolder = ReposIndexer.ExecutionDirectory + Path.DirectorySeparatorChar + repo.Id;
+            var repoCacheFile = repoFolder + CacheFileSuffix;
+            _logger.LogTrace("Saving cache for repo {RepoId} to file {FileName}", repo.Id, repoCacheFile);
+            File.WriteAllText(repoCacheFile, JsonConvert.SerializeObject(repo.Dependencies));
+        }
+
+        /// <summary>
+        /// Tries to read the cache file of a repository.
+        /// </summary>
+        /// <param name="repo">Repo to read the cache file for</param>
+        /// <param name="cached">The read cached version or null if none has been created.</param>
+        /// <returns>true if a cache file has been found and loaded.</returns>
+        public bool TryGetCachedVersion(WritableRepositoryInformation repo, out RepositoryInformation cached)
+        {
+            var repoFolder = ReposIndexer.ExecutionDirectory + Path.DirectorySeparatorChar + repo.Id;
+            var repoCacheFile = repoFolder + CacheFileSuffix;
+
+            _logger.LogTrace("Cache lookup for repo {RepoId}", repo.Id);
+            cached = null;
+            if (File.Exists(repoCacheFile))
+            {
+                _logger.LogTrace("Cache hit for repo {RepoId} on file {FileName}", repo.Id, repoCacheFile);
```

Comment:
This is pretty important since this drastically changes the behavior of the job. I would make this an information

```suggestion
                _logger.LogInformation("Cache hit for repo {RepoId} on file {FileName}", repo.Id, repoCacheFile);
```